### DISCUSSION TITLE
How to select a ban action for a specific jail (banaction vs banaction_allports)?

### DISCUSSION THREAD

**Discussion by paulvirtuel** [fail2ban/fail2ban#3673] [Q&A] [answered]
My setup on Debian Bookworm works but only bans the specific port(s) defined in the jail.  I would like to ban the IP, meaning all ports for some jails.

Say I have this in my jail.conf
~~~
[DEFAULT]
banaction = nftables
banaction_allports = nftables[type=allports]

[jail1]
enabled = true
port    = some port

[jail2]
enabled = true
port    = some port
~~~

Q1- I am wondering how I can ban the ip (all ports) for jail1 (e.g. sshd) and just the port for jail2 (e.g. ftp)?
Would that work?
~~~
[jail1]
banaction = nftables[type=allports]

[jail2]
banaction = nftables
~~~

Q2- I did not find an explanation in the wiki on how/when I can use banaction_allports?

**Comment by sebres** [comment, marked as answer]
> Q1- I am wondering how I can ban the ip (all ports) for jail1 (e.g. sshd) and just the port for jail2 (e.g. ftp)?
Would that work?

Yes. However, instead, I'd use `banaction = %(banaction_allports)s` for jail1 (so if default changes, it'd be changed too) and no `banaction` for jail2, since it is your default. You'd need only port/whatever there. 
Thus:
```ini
[DEFAULT]
banaction = nftables
banaction_allports = nftables[type=allports]

[jail1]
enabled = true
banaction = %(banaction_allports)s

[jail2]
enabled = true
port = 21
```

> Q2- I did not find an explanation in the wiki on how/when I can use banaction_allports?

Pythons ini-files supports string interpolation `%(banaction_allports)s` and we have our own enhancement `<banaction_allports>`.
[Man page jail.conf.5](https://manpages.debian.org/experimental/fail2ban/jail.conf.5.en.html#CONFIGURATION_FILES_FORMAT) definitely describes [this](https://manpages.debian.org/experimental/fail2ban/jail.conf.5.en.html#after).

**Reply by paulvirtuel** [reply]
Thanks!   I will do what you suggest.
For Q2, thanks for the link.  I was reading the github wiki and could not find it:
[https://github.com/fail2ban/fail2ban/wiki](url)

**Reply by sebres** [reply]
Wiki is just... wiki.
We don't duplicate the complete documentation there.
However possibly it'd be not bad to extend [wiki :: Proper fail2ban configuration](https://github.com/fail2ban/fail2ban/wiki/Proper-fail2ban-configuration) with more info about ini-file substitution.

**Reply by paulvirtuel** [reply]
Thanks again for your help.  
I understand, documentation is a lot of work.  Especially with so many linux distributions and configuration possibilities.
Maybe it is just me, but I usually look at github first.  On the plus side, a bigger wiki might save you from answering common questions.
